subsys/llext/Kconfig

```diff
@@ -65,6 +65,25 @@ config LLEXT_STORAGE_WRITABLE
 	  Select if LLEXT storage is writable, i.e. if extensions are stored in
 	  RAM and can be modified in place
 
+config LLEXT_EXPORT_BUILTINS_BY_SLID
+	bool "export built-in symbols to llexts via SLIDs"
+	help
+	  When enabled, symbols exported from the Zephyr kernel
+	  or application (via EXPORT_SYMBOL) are linked to LLEXTs
+	  via Symbol Link Identifiers (SLIDs) instead of name.
+
+	  Enabling this option provides a huge size reduction,
+	  makes the linking process faster and provides more
+	  confidentiality, as exported symbol names are dropped
+	  from the binary. However, it can make LLEXT debugging
+	  harder and prevents usage of 'llext_find_sym' to look
+	  up symbols from the built-in table by name. It also
+	  requires the LLEXTs to be post-processed after build.
+
+	  This option does not affect the behavior of linking
+	  symbols between multiple LLEXTs, only between LLEXTs
+	  and the Zephyr kernel / application.
```

Comment:
Linking between llexts is not supported at the moment, but it makes sense to keep this for the future if it's valid. However, it's not clear to me why linking between llexts would be any different than with the main Zephyr binary. Can you explain a bit more?

Reply:
I assumed that linking between LLEXTs had been implemented, for some reason. Mistake on my part.

As for why linking is different:
* linking to Zephyr is done by parsing the built-in export table, which is just a section filled with `struct llext_const_symbol`s
* linking to an LLEXT is done by parsing the ELF symbol table

There is however no huge technical difficulty to implement SLID linking for cross-LLEXT linking, but it is just not implemented by this PR so I wanted to make that clear (hence the option name, too).

Should this message be removed / reworded?

Reply:
Thanks for the explanation. Yes, there is currently no inter-llext linking and I'm not sure using the symbol table then is a good choice (I think it would easily cause mis-linking issue; I would vote for restricting to symbols in the export table). I would say this should be removed ATM.

Reply:
You are right, I had forgotten that LLEXTs also have an export table, which is indeed what is used by `llext_find_sym()` when looking up symbols:
https://github.com/zephyrproject-rtos/zephyr/blob/149df6b61b6b3baee686cd62c9eb92237ff58d84/subsys/llext/llext.c#L127-L132

where the `sym_table` parameter comes from `struct llext`'s `exp_tab` field, populated here from `.exported_sym` section:
https://github.com/zephyrproject-rtos/zephyr/blob/149df6b61b6b3baee686cd62c9eb92237ff58d84/subsys/llext/llext.c#L502-L531
(Nit: this code appears to create a copy of the export table, which sounds like a huge RAM waste? But that's out of this PR's scope)

So this part of my comment was wrong, but I don't think it invalidates the point I was trying to make, i.e., the built-in export table is treated differently from LLEXTs', hence allowing to implement SLID-based linking for one or the other independently.


I'll remove the message.